modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java

```diff
@@ -0,0 +1,213 @@
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.carbon.am.tests.sample;
+
+import org.json.JSONObject;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.am.tests.APIManagerIntegrationTest;
+import org.wso2.carbon.am.tests.util.APIPublisherRestClient;
+import org.wso2.carbon.am.tests.util.APIStoreRestClient;
+import org.wso2.carbon.am.tests.util.bean.APILifeCycleState;
+import org.wso2.carbon.am.tests.util.bean.APILifeCycleStateRequest;
+import org.wso2.carbon.am.tests.util.bean.APIRequest;
+import org.wso2.carbon.am.tests.util.bean.GenerateAppKeyRequest;
+import org.wso2.carbon.am.tests.util.bean.SubscriptionRequest;
+import org.wso2.carbon.automation.api.clients.user.mgt.UserManagementClient;
+import org.wso2.carbon.automation.core.utils.HttpRequestUtil;
+import org.wso2.carbon.automation.core.utils.HttpResponse;
+
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This covers adding two scopes with different roles and accessing them with users with those roles
+ * This Test-case uses the existing admin role and admin user
+ * Doc For API used :- http://sqlrest.sourceforge.net/5-minutes-guide.htm
+ */
+public class APIScopeTestCase extends APIManagerIntegrationTest {
+
+    private APIPublisherRestClient apiPublisher;
+    private APIStoreRestClient apiStore;
+    private String publisherURLHttp;
+    private String storeURLHttp;
+
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init(0);
+        if (isBuilderEnabled()) {
+            publisherURLHttp = getServerURLHttp();
+            storeURLHttp = getServerURLHttp();
+
+        } else {
+            publisherURLHttp = getPublisherServerURLHttp();
+            storeURLHttp = getStoreServerURLHttp();
+        }
+        apiPublisher = new APIPublisherRestClient(publisherURLHttp);
+        apiStore = new APIStoreRestClient(storeURLHttp);
+
+    }
+
+    @Test(groups = {"wso2.am"}, description = "Testing the scopes with admin, subscriber roles")
+    public void testSetScopeToResourceTestCase() throws Exception {
+
+
+        UserManagementClient userManagementClient = new UserManagementClient(amServer.getBackEndUrl(), "admin",
+                                                                             "admin");
+
+        // adding new role subscriber
+        userManagementClient.addRole("subscriber", new String[]{}, new String[]{"/permission/admin/login",
+                                                                                "/permission/admin/manage/api/subscribe"});
+
+        // crating user john
+        userManagementClient.addUser("john", "john123", new String[]{"subscriber"}, "user_role");
+
+
+        // Adding API
```

Comment:
Variable names should start with a simple case letter and follow camel case naming style.